src/threedidepth/calculate.py

```diff
@@ -605,6 +592,48 @@ def __call__(self, progress):
         )
 
 
+class ResultAdmin:
+    """
+    args:
+        gridadmin_path (str): Path to gridadmin.h5 file.
+        results_3di_path (str): Path to (aggregate_)results_3di.nc file.
+
+    Wraps either GridH5ResultAdmin or GridH5AggregateResultAdmin, based on the
+    result type of the file at results_3di_path . Also has custom properties
+    `result_type`, `variable` and `calculation_steps`.
+    """
+
+    def __init__(self, gridadmin_path, results_3di_path):
+        with h5py.File(results_3di_path) as h5:
+            self.result_type = h5.attrs['result_type'].decode('ascii')
+            self.variable = {
+                "raw": "s1", "aggregate": "s1_max",
+            }[self.result_type]
+            self.calculation_steps = len(h5['Mesh1D_' + self.variable])
+
+        result_admin_args = gridadmin_path, results_3di_path
+        if self.result_type == "raw":
+            self._result_admin = GridH5ResultAdmin(*result_admin_args)
+        else:
+            self._result_admin = GridH5AggregateResultAdmin(*result_admin_args)
+
+    def get_timestamps(self, calculation_steps):
+        if self.result_type == "raw":
+            return self.nodes.timestamps[calculation_steps]
```

Comment:
Krijg je met nodes.timestamps niet alleen de eerste 10 bij default?

Reply:
Ik hoop het niet! In elk geval krijg ik er bij mijn testbestand 13 :-)

Reply:
Okido, just checking ;)